content/docs/guide.md

```diff
@@ -1,94 +1,301 @@
+project settings. In the left sidebar, click "Integration & services" and you
+will see the application displayed there.
+
+
+### Staging instance
+
+[Packit-as-a-Service-stg](https://github.com/apps/packit-as-a-service-stg) GitHub App
+runs the latest code. It can be used to verify latest changes until they get
+(every Tuesday) into the [production instance](https://github.com/marketplace/packit-as-a-service).
+If you want to enjoy the freshest features and help us with the development,
+see more details [here](https://github.com/packit/packit/discussions/1530).
+
+![Packit GitHub application: Staging instance](/images/guide/guide_github_app_stg.png)
+
+
+### GitLab
+
+GitLab support is mainly experimental and is influenced by the presence of CentOS Stream source-git workflow.
+**We do not verify** the functionality of the GitLab support on the day-to-day basis as we do with the GitHub App.
+
+
+#### How to set up Packit on GitLab
+
+GitLab doesn't have an app functionality and we don't use the [Integrations](https://docs.gitlab.com/ee/user/project/integrations/overview.html) as it doesn't solve authentication.
+To enable Packit, you need to manually configure a webhook.
+
+1. Go to `Settings → Webhooks` and add https://prod.packit.dev/api/webhooks/gitlab or https://stg.packit.dev/api/webhooks/gitlab as the `URL`.
+   ![GitLab webhooks](/images/gitlab/webhooks.png)
+2. Select `Push events`, `Comments`, `Issues events`, `Merge request events` and `Release events` as the `Triggers`.
+   Do not provide `Secret token` yet and click `Add webhook`.
+   You should see the created hook at the bottom of the page now.
+   ![Setting up webhook](/images/gitlab/set_up_webhook.png)
+3. Select `Test` and for example `Merge requests events`.
+   At the top of the page appears a red `Hook executed successfully but returned HTTP 401 "X-Gitlab-Token not in request.headers"`.
+4. Wait a bit and check `Issues` of the project and after a while there should be
+   a new confidential `Packit-Service Authentication` issue with the instructions how to finish the setup.
+
+##### GitLab Pipelines
+
+With some limitations (but tweakability) you can also use Packit as a GitLab pipeline
+using the Packit's [CLI](/docs/cli) from the regularly built
+[`quay.io/packit/packit` image](https://quay.io/repository/packit/packit).
+As you can see in the following example, it's really straightforward and explicit:
+
+```yaml
+image: quay.io/packit/packit
+
+srpm_build:
+  stage: build
+  script:
+  - packit srpm
+  artifacts:
+    paths:
+      - "*.src.rpm"
+    expire_in: 1 week
+```
+
+### Pagure
+
+We have rudimentary support for the Pagure instances since they are used in the downstream workflow.
+Packit cannot be set up as a CI (in the same sense as on GitHub or GitLab) on Pagure instances,
+since the demand for this feature is not significant.
+The differences between GitHub/GitLab and Pagure APIs would result in many workarounds to be implemented, if possible at all.
```

Comment:
```suggestion
The differences between GitHub/GitLab and Pagure APIs would result in many workarounds having to be implemented, if it would be possible at all.
```